q2_demux/_filter.py

```diff
@@ -0,0 +1,42 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import pandas as pd
+
+from qiime2 import Metadata
+from qiime2.util import duplicate
+from q2_demux._summarize import _PlotQualView
+from q2_types.per_sample_sequences import \
+    CasavaOneEightSingleLanePerSampleDirFmt
```

Comment:
extreme nitpicking:

Put the q2_demux import at the bottom (with a blank line separating it). This makes it quick to find imports to ourself